src/features/tutorials/components/tutorial-form/index.tsx

```diff
@@ -0,0 +1,181 @@
+import { Button } from '@chakra-ui/react';
+import { useForm } from 'react-hook-form';
+import { useMemo, useRef, useState } from 'react';
+import { GrDocumentPdf } from 'react-icons/gr';
+import { ControlledSelect, Input } from '@/components/form-fields';
+import { TutorialPayload, File } from '../../type';
+import { useGetAllCategoryTutorial } from '@/features/categories-tutorial/api/get-all-categories-tutorial';
+
+interface TutorialFromProps {
+  defaultValues?: TutorialPayload;
+  onSubmit: (data: TutorialPayload) => void;
+  isSubmitting: boolean;
+}
+
+export function TutorialForm({
+  defaultValues,
+  onSubmit,
+  isSubmitting,
+}: TutorialFromProps) {
+  const isEditing = useMemo(() => Boolean(defaultValues), [defaultValues]);
+  let nomeArquivo;
+  const {
+    register,
+    handleSubmit,
+    control,
+    formState: { errors },
+  } = useForm({
+    defaultValues: {
+      ...defaultValues,
+    },
+  });
+
+  const { data: categories, isLoading: isLoadingCategories } =
+    useGetAllCategoryTutorial();
+
+  const categoryOptions = categories?.map((category) => ({
+    label: category?.name,
+    value: category?.id,
+  }));
+
+  const [fileName, setFileName] = useState<File>();
+  const inputRef = useRef();
+
+  const handleDragOver = (event: any) => {
+    event.preventDefault();
+  };
+  const handleDrop = (event: any) => {
+    event.preventDefault();
+    // console.log(event.dataTransfer.files[0]);
+    setFileName(event.dataTransfer.files[0]);
+  };
+
+  if (fileName) {
+    nomeArquivo = fileName.name;
+  }
+
+  return (
+    <form onSubmit={handleSubmit(onSubmit)} encType="multipart/form-data">
+      <Input
+        label="Nome do tutorial"
+        {...register('name', { required: 'Campo obrigatório' })}
+        placeholder="Digite o nome do tutorial"
+        errors={errors?.name}
+      />
+
+      {/* <Input
+        label="ID da categoria"
+        {...register('category_id', { required: 'Campo obrigatório' })}
+        placeholder="Digite o id da categoria"
+        errors={errors?.name}
+      /> */}
+
+      <ControlledSelect
+        control={control}
+        name="category_id"
+        id="category_id"
+        options={categoryOptions}
+        isLoading={isLoadingCategories}
+        placeholder="Categoria"
+        label="Categoria"
```

Comment:
Precisa adicionar essa regra para avisar que o campo é obrigatório.
```suggestion
        label="Categoria"
        rules={{ required: 'Campo obrigatório.' }}
```